docs/datatypes/chain.md

```diff
@@ -1,60 +1,70 @@
 
 `List` is a great data type, it is very simple and easy to understand.
-It has very low overhead for the most important functions such as `fold` and `map` and also supports prepending a single element in constant time.
+It has very low overhead for the most important functions such as [fold][Foldable] and [map][Functor] and also supports prepending a single element in constant time.
 
-Traversing a data structure with something like `Writer[List[Log], A]` or `ValidatedNel[Error, A]` is  powerful and allows us to precisely specify what kind of iteration we want to do while remaining succint.
+Traversing a data structure with something like [Writer\[List\[Log\], A\]][Writer] or [ValidatedNel\[Error, A\]][Validated] is powerful and allows us to precisely specify what kind of iteration we want to do while remaining succinct.
 However, in terms of efficiency it's a whole different story unfortunately.
-That is because both of these traversals make use of the `List` monoid (or the `NonEmptyList` semigroup), which by the nature of `List` is very inefficient.
-If you use `traverse` with a data structure with `n` elements and `Writer` or `Validated` as the `Applicative` type, you will end up with a runtime of `O(n^2)`.
+That is because both of these traversals make use of the `List` monoid (or the [NonEmptyList] semigroup), which by the nature of `List` is very inefficient.
+If you use [traverse][Traverse] with a data structure with `n` elements and [Writer] or [Validated] as the [Applicative] type, you will end up with a runtime of `O(n^2)`.
 This is because, with `List`, appending a single element requires iterating over the entire data structure and therefore takes linear time.
 
-So `List` isn't all that great for this use case, so let's use `Vector` or `NonEmptyVector` instead, right?
+So @:api(scala.collection.immutable.List) isn't all that great for this use case, so let's use @:api(scala.collection.immutable.Vector) or @:api(cats.data.NonEmptyVector)` instead, right?
 
 Well, `Vector` has its own problems and in this case it's unfortunately not that much faster than `List` at all. You can check [this blog post](http://www.lihaoyi.com/post/BenchmarkingScalaCollections.html#vectors-are-ok) by Li Haoyi for some deeper insight into `Vector`'s issues.
 
 
 `Chain` evolved from what used to be `fs2.Catenable` and Erik Osheim's [Chain](https://github.com/non/chain ) library.
 Similar to `List`, it is also a very simple data structure, but unlike `List` it supports both constant O(1) time `append` and `prepend`.
-This makes its `Monoid` instance super performant and a much better fit for usage with `Validated`,`Writer`, `Ior` or `Const`.
+This makes its [Monoid] instance [super performant][Benchmarks] and a much better fit for usage with [Validated], [Writer], [Ior] or [Const].
 
-To utilize this Cats includes type aliases like `ValidatedNec` or `IorNec` as well as helper functions like `groupByNec` or `Validated.invalidNec`.
```

Comment:
I can't find any definition for `groupByNec`??

Reply:
https://github.com/typelevel/cats/blob/708cbaba2e6ac07a5ac1819b6c18b0c83a2faaba/core/src/main/scala/cats/syntax/list.scala#L167-L170

Reply:
That's really awkward. There is no scaladoc generated for that. The closest I can get is here, but there is no doc for the `ListOpsBinCompat0` class: https://www.javadoc.io/static/org.typelevel/cats-docs_2.13/2.9.0/cats/implicits$.html#catsSyntaxListBinCompat0[A](la:List[A]):cats.syntax.ListOpsBinCompat0[A]

Reply:
Yeah 😕 if there's no scaladoc, that also means there's also no bincompat support from MiMa. Also I assume the fact that these private classes can define methods that are publicly usable is a manifestation of https://github.com/scala/bug/issues/6794.

Suffice to say, I've been making these public.

Reply:
Yikes. Is flipping that to public just the one line change? I can expose that as part of the PR if it's not going to scope creep too much.

Reply:
Yeah, this is just a systemic issue across the codebase, we can deal with it elsewhere.